plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/errors/VerificationError.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.viper.errors
+
+import org.jetbrains.kotlin.formver.viper.ast.Position
+import viper.silver.verifier.errors
+
+
+sealed interface VerificationError : VerifierError {
+    val result: viper.silver.verifier.AbstractVerificationError
+    override val id: String
+        get() = result.id()
+    override val msg: String
+        get() = result.toString()
+    override val position: Position
+        get() = Position.fromSilver(result.pos())
+}
+
+data class PreconditionInCallFalse(
+    override val result: errors.PreconditionInCallFalse,
+) : VerificationError
+
+data class PostconditionViolated(
+    override val result: errors.PostconditionViolated,
+) : VerificationError {
+    override val position: Position
+        get() = Position.fromSilver(result.member().pos())
+}
+data class AssertFailed(
+    override val result: errors.AssertFailed,
+) : VerificationError
```

Comment:
I'm a bit confused about what exactly this translation gets us.  We want failures to be reported in terms of the Kotlin code, but this just re-wraps the Viper errors.  Is the goal just re-wrapping the position, or is there more to it?

Reply:
I created these translation classes as adapters between Scala and Kotlin. So we do not have to worry about Scala's implementation details (acting as abstraction layer). At the moment no special functions are defined, but they will be when I start working on better error messages.